File: articles/from-ideas-to-strategy/c/strategy_conversion_score.c
/*
Low-level idea-to-strategy conversion score utilities.
Compile: cc strategy_conversion_score.c -o strategy_conversion_score
Run: ./strategy_conversion_score
*/

#include <stdio.h>

typedef struct {
    const char *name;
    double feasibility;
    double viability;
    double desirability;
    double integration_difficulty;
    double execution_readiness;
    double strategic_fit;
    double evidence_confidence;
    double ethical_resilience;
} Initiative;

double conversion_score(Initiative i) {
    return 0.16 * i.feasibility +
           0.18 * i.viability +
           0.16 * i.desirability -
           0.12 * i.integration_difficulty +
           0.16 * i.execution_readiness +
           0.12 * i.strategic_fit +
           0.08 * i.evidence_confidence +
           0.06 * i.ethical_resilience;
}

int main(void) {
    Initiative initiatives[] = {
        {"High-Idea Low-Execution Concept", 0.78, 0.38, 0.82, 0.72, 0.29, 0.62, 0.42, 0.54},
        {"Balanced Strategic Initiative", 0.74, 0.79, 0.77, 0.44, 0.81, 0.84, 0.76, 0.72},
        {"Integration-Challenged Initiative", 0.69, 0.71, 0.84, 0.83, 0.52, 0.76, 0.58, 0.60}
    };

    int count = sizeof(initiatives) / sizeof(initiatives[0]);

    for (int i = 0; i < count; i++) {
        double score = conversion_score(initiatives[i]);
        printf("%s | score %.3f | confidence-adjusted %.3f\n",
               initiatives[i].name,
               score,
               score * initiatives[i].evidence_confidence);
    }

    return 0;
}

File: articles/from-ideas-to-strategy/rust/strategy_conversion_score.rs
// Command-line idea-to-strategy conversion scoring scaffold.
// Compile: rustc strategy_conversion_score.rs -o strategy_conversion_score
// Run: ./strategy_conversion_score

#[derive(Debug)]
struct Initiative {
    name: &'static str,
    feasibility: f64,
    viability: f64,
    desirability: f64,
    integration_difficulty: f64,
    execution_readiness: f64,
    strategic_fit: f64,
    evidence_confidence: f64,
    ethical_resilience: f64,
}

fn conversion_score(i: &Initiative) -> f64 {
    0.16 * i.feasibility
        + 0.18 * i.viability
        + 0.16 * i.desirability
        - 0.12 * i.integration_difficulty
        + 0.16 * i.execution_readiness
        + 0.12 * i.strategic_fit
        + 0.08 * i.evidence_confidence
        + 0.06 * i.ethical_resilience
}

fn main() {
    let initiatives = vec![
        Initiative { name: "High-Idea Low-Execution Concept", feasibility: 0.78, viability: 0.38, desirability: 0.82, integration_difficulty: 0.72, execution_readiness: 0.29, strategic_fit: 0.62, evidence_confidence: 0.42, ethical_resilience: 0.54 },
        Initiative { name: "Balanced Strategic Initiative", feasibility: 0.74, viability: 0.79, desirability: 0.77, integration_difficulty: 0.44, execution_readiness: 0.81, strategic_fit: 0.84, evidence_confidence: 0.76, ethical_resilience: 0.72 },
        Initiative { name: "Integration-Challenged Initiative", feasibility: 0.69, viability: 0.71, desirability: 0.84, integration_difficulty: 0.83, execution_readiness: 0.52, strategic_fit: 0.76, evidence_confidence: 0.58, ethical_resilience: 0.60 },
    ];

    for i in initiatives {
        let score = conversion_score(&i);
        println!("{} | score {:.3} | confidence-adjusted {:.3}", i.name, score, score * i.evidence_confidence);
    }
}
